ui/app/components/customize-gas-modal/index.js

```diff
@@ -97,7 +99,11 @@ inherits(CustomizeGasModal, Component)
 function CustomizeGasModal (props) {
   Component.call(this)
 
-  this.state = getOriginalState(props)
+  const originalState = getFreshState(props)
+  this.state = {
+    ...originalState,
+    originalState,
```

Comment:
Is this needed?

Reply:
I think it is needed to maintain current "revert" functionality